### PR TITLE
Update crown for pool elements

### LINKED ISSUE #819: Update pool to use new crown
**Acceptance Criteria:**
Should have add-lane-above, add-lane-bottom, sequence flow, and trash icon

Screen Shot 2019-11-14 at 12 39 43 PM

Covers: 
- Pool

Related: #789

### PR DESCRIPTION
**Pull request by chipit24**
Fixes #819.

### CODE REVIEW

File: src/mixins/resizeConfig.js [hunk 1]
```diff
@@ -486,7 +482,7 @@ export default {
     },
     setPointAttributes(point, cursorDirection) {
       point.attr({
-        root: { display: 'none' },
+        root: { display: 'initial' },
```
Comment (chipit24) [inline src/mixins/resizeConfig.js]: This was causing the "resize squares" to not display initally when the pool was dragged onto the grid. Changing it to `'initial'` fixes this.

File: src/components/crownConfig.vue [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <template>
   <div class="crown-config" :style="style" v-if="showCrown">
+    <slot/>
```
Comment (chipit24) [inline src/components/crownConfig.vue]: Decided to introduce a slot to add component specific buttons (in this case, the add lane buttons).

File: src/components/crownConfig.vue [hunk 3]
```diff
@@ -204,7 +182,25 @@ export default {
...
+      const { x, y, width } = shapeView.getBBox();
+
+      this.style = {
+        top: `${y - 45}px`,
+        left: `${x + width - 20}px`,
+        cursor: 'pointer',
+      };
+    },
     setUpCrownConfig() {
+      this.paper.on('render:done scale:changed translate:changed', this.repositionCrown);
```
Comment (chipit24) [inline src/components/crownConfig.vue]: I triggered custom events on the paper to handle certain changes that required repositioning the crown. I found this allowed me to get an actual width from `getBBox`.

File: src/components/nodes/pool/pool.vue [hunk 4]
```diff
@@ -1,23 +1,51 @@
...
+    :nodeRegistry="nodeRegistry"
+    :moddle="moddle"
+    :collaboration="collaboration"
+    :process-node="processNode"
+    :plane-elements="planeElements"
+    :is-rendering="isRendering"
+    @remove-node="$emit('remove-node', $event)"
+    @add-node="$emit('add-node', $event)"
+    @save-state="$emit('save-state', $event)"
+  >
```
Comment (lisa-fehr) [inline src/components/nodes/pool/pool.vue]: Pool has allowSetNodePosition which is in the crown component.
Does `element.component.allowSetNodePosition` in this file still work with the new crown component?

File: src/components/crownConfig.vue [hunk 5]
```diff
@@ -204,7 +182,25 @@ export default {
 
       this.$emit('save-state');
     },
+    repositionCrown() {
+      const shapeView = this.shape.findView(this.paper);
+
+      if (!shapeView) {
```
Comment (lisa-fehr) [inline src/components/crownConfig.vue]: Boundary events and intermediate events are now placing the crown at the end of the label on initial drop.

Comment (lisa-fehr) [reply src/components/crownConfig.vue]: If you drag an escalation event from a call activity to a task, the crown moves to the label after error.

File: src/components/nodes/pool/pool.vue (hunk 4 above)
Comment (chipit24) [reply src/components/nodes/pool/pool.vue]: Yup, that still seems to work as expected. Did you notice any problems with it?

File: src/components/crownConfig.vue (hunk 5 above)
Comment (chipit24) [reply src/components/crownConfig.vue]: Looks like there's a `useModelGeometry` option that can be passed to `getBBox` that fixed this:
https://resources.jointjs.com/docs/jointjs/v3.1/joint.html#dia.ElementView.prototype.getBBox

That's super useful!

**Comment by chipit24** [comment]
Adding WIP label as I'm going to pull out logic related to `allowSetNodePosition` to fix the interaction with the pool when an element is dragged outside of it.

The current issue is that the pool component reaches into the component state of an element that is being dragged off of it, and sets that component's `allowSetNodePosition` value to `false`.

The `allowSetNodePosition` value was initially mixed into all components, but now resides in the new crown-config component that is a child component. So fixing this directly would require passing down the `allowSetNodePosition` prop.

There are a few issues with this: 
* This should not be logic that is in the `crownConfig` file.
* Reaching into a component's internal state should be avoided.
* "Prop drilling" (components basically acting as middlemen for props and passing them down without using the prop themselves) should be avoided.

**Pull request merged**